### ISSUE TITLE
prefer-stateless-function documentation is wrong

### ISSUE THREAD

**Issue by MichalZalecki** [yannickcr/eslint-plugin-react#588]
Hi!

I'm facing false positive for rule `prefer-stateless-function` with latest dependencies.

![image](https://cloud.githubusercontent.com/assets/3136577/15075326/10721528-13a4-11e6-9f03-26019593d89a.png)

```js
// .eslintrc.json
{
  "extends": "airbnb",
  "parser": "babel-eslint",
  "settings": {
    "import/resolver": {
      "webpack": { "config": "./webpack/webpack.prod.config.js" }
    }
  },
  "env": {
    "jasmine": true
  },
  "rules": {
    "quotes": ["error", "double"],
    "no-unused-vars": [2, { "vars": "local", "args": "after-used", "argsIgnorePattern": "^_" }]
  }
}
```

```js
// package.json
"babel-eslint": "^6.0.4",
"eslint": "^2.9.0",
"eslint-config-airbnb": "^8.0.0",
"eslint-import-resolver-webpack": "^0.2.4",
"eslint-plugin-import": "^1.6.1",
"eslint-plugin-jsx-a11y": "^1.0.4",
"eslint-plugin-react": "^5.0.1",
```

**Comment by philiiiiiipp** [comment]
I have the same issue

```
"eslint": "^2.8.0",
"eslint-config-airbnb": "^8.0.0",
"eslint-plugin-import": "^1.6.1",
"eslint-plugin-jsx-a11y": "^1.0.3",
"eslint-plugin-react": "^5.0.1",
```

eslint.rc:
```
{
  "eslintignore": [
    "build",
    "node_modules"
  ],
  "parser": "babel-eslint",
  "extends": "airbnb",
  "globals": {
    "__DEV__": true
  },
  "env": {
    "browser": true
  },
  "rules": {
    "indent": [
      "warn",
      2
    ],
    "comma-dangle": [
      "error",
      "never"
    ],
    "arrow-body-style": [
      "error",
      "always"
    ],
    "no-confusing-arrow": 0,
    "react/jsx-quotes": 0,
    "jsx-quotes": [
      2,
      "prefer-single"
    ]
  }
}
```
The file in question: 
```
export default class App extends Component {
  render() {
    return (
      <Router routes={routes} history={browserHistory} />
    );
  }
}
```

**Comment by ljharb** [comment]
These are not false positives. A class-based component that has nothing but a render method should be an SFC instead.

**Comment by philiiiiiipp** [comment]
Ah true, I see. Should have looked more closely at the explanation *ashamed*

**Comment by ljharb** [comment]
@MichalZalecki i'm going to close this, but will be happy to reopen if further clarification is needed

**Comment by MichalZalecki** [comment]
@ljharb According to this doc, yes https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-stateless-function.md

Look at "The following patterns are not considered warnings:". 2. example is failing on that rule.

**Comment by ljharb** [comment]
Hmm, you're right. I think the documentation is wrong then.

**Comment by ljharb** [comment]
@MichalZalecki ah, i think i get the difference - in React 15, SFCs *can* return null. So, that example should only not be a warning in React 0.13 or 0.14.
